prow/github/client.go

```diff
@@ -636,6 +642,8 @@ type ClientOptions struct {
 	AppID         string
 	AppPrivateKey func() *rsa.PrivateKey
 
+	IsGitHubAction bool
```

Comment:
Basically the same quesiton as chao, but I can not unresolve the other conversation.

From the [docs](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) it seems there are a couple of envionment variables, for example this one seems well-suited:

> GITHUB_ACTIONS Always set to true when GitHub Actions is running the workflow. You can use this variable to differentiate when tests are being run locally or by GitHub Actions.

Reply:
Yes, we could check for that env var and then that tightly couples this module to an implementation detail in GIthub Actions which I'd prefer to avoid. 

By adding a field we leave it up to the instantiator of this class at compile time instead of env var lookups at run time. 

Given that the knowledge of whether or not the client is destined to run in a GH Action is known at compile time I think this is the better choice.

The field defaults to false which would suit the majority of current Github Client use cases and removes the need to do a runtime lookup on the env.

Are there any other considerations or use cases that I'm not taking into account? On my phone right now. I'll check this again when I get to a desk.

Reply:
Well this change is tightly coupled to github actions behavior so I don't see why the lookup would make this any worse. I am okay leaving the flag, but its default value should be derived from the presence of that env var IMHO

Reply:
Bumped into Robert in person, had a quick chat about this, also update here: on the same page as Alvaro, this is not a super big deal, the entire idea is to reduce the friction of using this tool by avoiding tons of command line args. So yeah set the default flag value as `os.GetEnv("GITHUB_ACTIONS")` makes sense to me